### ISSUE TITLE
`no-unused-array-method-return` false positive for type assertion

### ISSUE THREAD

**Issue by FloEdelmann** [sindresorhus/eslint-plugin-unicorn#3064]
`no-unused-array-method-return` already ignores variables whose type is known not to be an array. But it doesn't ignore type assertions:

```ts
interface Foo {
    filter: () => void
}

declare const foo: Foo
declare const bar: any

foo.filter()           // correctly not reported
;(bar as Foo).filter() // wrongly reported → false-positive
```

**Comment by FloEdelmann** [comment]
I have a working fix in https://github.com/FloEdelmann/eslint-plugin-unicorn/tree/no-unused-array-method-return-assertions / https://github.com/FloEdelmann/eslint-plugin-unicorn/commit/b76d0815917681ad1941d94c87d53f74c605f078.

Unfortunately, due to https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2975, I can't open a PR, even though I have contributed in the past: https://github.com/sindresorhus/eslint-plugin-unicorn/commits?author=FloEdelmann
